helper/builtinplugins/builtin.go

```diff
@@ -0,0 +1,30 @@
+package builtinplugins
+
+import (
+	"github.com/hashicorp/vault/plugins/database/mysql"
+	"github.com/hashicorp/vault/plugins/database/postgresql"
+)
+
+type BuiltinFactory func() (interface{}, error)
```

Comment:
Mainly just for extendability, all builtin plugins for this, and potentially future, backends will use this factory function. It's possible some of those might need to return errors.